src/layouts/sidebar-sidecar/index.tsx

```diff
@@ -1,12 +1,13 @@
-import { ReactElement } from 'react'
+import { ReactElement, useEffect, useMemo, useState } from 'react'
```

Comment:
The code added here is long, plan is to move it to helper functions if we like the approach so the file isn't so long.

Reply:
Makes sense. Would this helper function then be used by other layouts / views?

Reply:
No, the main helper I'm thinking of would be one that initializes `finalSidebarProps`. The reusable helpers that I think views could use though are the two new ones in `sidebar/helpers`:

- `generateProductLanding`
- `generateTopLevelNavItems`

What do you think?